BCnEnc.Net/Decoder/BcDecoder.cs

```diff
@@ -1,552 +1,706 @@
+		{
+			return DecodeRawInternal(input, format, pixelWidth, pixelHeight, default);
+		}
+
+		/// <summary>
+		/// Read a Ktx or a Dds file from a stream and decode it.
+		/// </summary>
+		/// <param name="inputStream">The stream containing a Ktx or Dds file.</param>
+		/// <returns>The decoded Rgba32 image.</returns>
+		public Image<Rgba32> Decode(Stream inputStream)
+		{
+			return Decode(inputStream, false, default)[0];
+		}
+
+		/// <summary>
+		/// Read a Ktx or a Dds file from a stream and decode it.
+		/// </summary>
+		/// <param name="inputStream">The stream containing a Ktx or Dds file.</param>
+		/// <returns>An array of decoded Rgba32 images.</returns>
+		public Image<Rgba32>[] DecodeAllMipMaps(Stream inputStream)
+		{
+			return Decode(inputStream, true, default);
+		}
+
+		/// <summary>
+		/// Read a Ktx file and decode it.
+		/// </summary>
+		/// <param name="file">The loaded Ktx file.</param>
+		/// <returns>The decoded Rgba32 image.</returns>
+		public Image<Rgba32> Decode(KtxFile file)
+		{
+			return Decode(file, false, default)[0];
+		}
+
+		/// <summary>
+		/// Read a Ktx file and decode it.
+		/// </summary>
+		/// <param name="file">The loaded Ktx file.</param>
+		/// <returns>An array of decoded Rgba32 images.</returns>
+		public Image<Rgba32>[] DecodeAllMipMaps(KtxFile file)
+		{
+			return Decode(file, true, default);
+		}
+
+		/// <summary>
+		/// Read a Dds file and decode it.
+		/// </summary>
+		/// <param name="file">The loaded Dds file.</param>
+		/// <returns>The decoded Rgba32 image.</returns>
+		public Image<Rgba32> Decode(DdsFile file)
+		{
+			return Decode(file, false, default)[0];
+		}
+
+		/// <summary>
+		/// Read a Dds file and decode it.
+		/// </summary>
+		/// <param name="file">The loaded Dds file.</param>
+		/// <returns>An array of decoded Rgba32 images.</returns>
+		public Image<Rgba32>[] DecodeAllMipMaps(DdsFile file)
+		{
+			return Decode(file, true, default);
+		}
+
+		#endregion
+
+		/// <summary>
+		/// Load a KTX or DDS file from a stream and extract either the main image or all mip maps.
+		/// </summary>
+		/// <param name="inputStream">The input stream to decode.</param>
+		/// <param name="allMipMaps">If all mip maps or only the main image should be decoded.</param>
+		/// <param name="token">The cancellation token for this operation. Can be default, if the operation is not asynchronous.</param>
+		/// <returns>An array of decoded Rgba32 images.</returns>
+		private Image<Rgba32>[] Decode(Stream inputStream, bool allMipMaps, CancellationToken token)
+		{
+			var position = inputStream.Position;
+			try
+			{
+				// Detect if file is a KTX or DDS by file extension
+				if (inputStream is FileStream fs)
+				{
+					var extension = Path.GetExtension(fs.Name).ToLower();
+					switch (extension)
+					{
+						case ".dds":
+							var ddsFile = DdsFile.Load(inputStream);
+							return Decode(ddsFile, allMipMaps, token);
+
+						case ".file":
```

Comment:
Here is ".file". Should be ".ktx". Maybe some auto-refactoring changed it?

Reply:
Yh, ReSharper probably caught it when I renamed something. I will change it.